saritasa_s3_tools/django/model_fields.py

```diff
@@ -42,6 +57,52 @@ def generate_filename(
         )
         return self.storage.generate_filename(filename)  # type: ignore
 
+    def _validate_file_existence(
+        self,
+        value: files.FieldFile | str,
+    ) -> None:
+        """Check key is present in bucket."""
+        is_field_file = isinstance(value, files.FieldFile)
+        file_obj = getattr(value, "_file", None)
+        if is_field_file and file_obj:
+            # Skip if value has reference to actual file
+            return  # pragma: no cover
+        try:
+            if not default_storage.exists(str(value)):
+                raise exceptions.ValidationError(
+                    _("File does not exist."),
+                )
+        # If url comes not from s3, then botocore on key validation will
+        # raise ParamValidationError
+        except botocore.exceptions.ParamValidationError as error:
+            raise exceptions.ValidationError(
+                error,
+            ) from error  # pragma: no cover
+        except exceptions.SuspiciousFileOperation as error:
+            raise exceptions.ValidationError(
+                error,
+            ) from error  # pragma: no cover
+
+    def _validate_key(
+        self,
+        value: files.FieldFile | str,
+    ) -> None:
+        """Check key is present in bucket."""
+        is_field_file = isinstance(value, files.FieldFile)
+        file_obj = getattr(value, "_file", None)
+        if is_field_file and file_obj:
+            # Skip if value has reference to actual file
+            # It will be saved with key generated by config
+            return  # pragma: no cover
+        if not self.s3_config:
+            raise exceptions.ImproperlyConfigured(  # pragma: no cover
+                "Please set s3_config for field",
+            )
+        if not self.s3_config.key.validate(str(value)):
+            raise exceptions.ValidationError(
+                _("File's path doesn't match with config's pattern."),
+            )
```

Comment:
```suggestion
        if is_field_file and file_obj:
            # Skip if value has reference to actual file
            # It will be saved with key generated by config
            return  # pragma: no cover

        if not self.s3_config:
            raise exceptions.ImproperlyConfigured(  # pragma: no cover
                "Please set s3_config for field",
            )

        if not self.s3_config.key.validate(str(value)):
            raise exceptions.ValidationError(
                _("File's path doesn't match with config's pattern."),
            )
```